### PR TITLE
feat: Support autoSelectFamily when connecting.

### LINKED ISSUE #1531: fetch times out in under 5 seconds
## Bug Description

When trying to fetch a URL a `fetch failed` error is thrown with the code as
    code: 'UND_ERR_CONNECT_TIMEOUT'
, This error is thrown at a request that barely takes 5 - 6 seconds to complete and other HTTP clients like axios and curl perform flawlessly on the same server

## Reproducible By

Fetch any discord API url
## Expected Behavior

The fetch should complete which is well below the 120s timeout

## Logs & Screenshots
```
/home/arnav/Documents/tej.js/node_modules/undici/lib/fetch/index.js:197
        Object.assign(new TypeError('fetch failed'), { cause: response.error })
                      ^
TypeError: fetch failed
    at Object.processResponse (/home/arnav/Documents/tej.js/node_modules/undici/lib/fetch/index.js:197:23)
    at /home/arnav/Documents/tej.js/node_modules/undici/lib/fetch/index.js:930:38
    at node:internal/process/task_queues:141:7
    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:138:8)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  cause: ConnectTimeoutError: Connect Timeout Error
      at Timeout.onConnectTimeout [as _onTimeout] (/home/arnav/Documents/tej.js/node_modules/undici/lib/core/connect.js:108:24)
      at listOnTimeout (node:internal/timers:561:11)
      at processTimers (node:internal/timers:502:7) {
    code: 'UND_ERR_CONNECT_TIMEOUT'
  }
}
```
## Environment

Gentoo, Node v17.6.0
### Additional context

Happen on every discord API url and used to work some time back but suddenly just fails everywhere.

### PR DESCRIPTION
**Pull request by ShogunPanda**
Fixes #1531.

This also unifies Node version detecting.

### CODE REVIEW

File: docs/api/Client.md
Comment (ronag) [inline docs/api/Client.md:32]: What's the default?

Comment (ShogunPanda) [reply docs/api/Client.md:32]: For `autoSelectFamily` is none (which is equivalent for to false).
For `autoSelectFamilyAttemptTimeout` is 250, but in Node, not in undici. I simply forward the option to node and that's why I haven't mentioned it here.

Comment (ronag) [reply docs/api/Client.md:32]: Should we maybe enable it as default? @mcollina

**Review by mcollina — COMMENTED** [review]
I would add a note in the README on how to overcome this problem.

I'm not convinced we should ship a change to this version by in a minor release. I would prefer to ship it as an option first, and then change the default once we bump the major.

**Review by ronag — APPROVED** [review]

File: docs/api/Client.md
Comment (ronag) [reply docs/api/Client.md:32]: @ShogunPanda then at least mention it uses the defaults from node otherwise?

Comment (ShogunPanda) [reply docs/api/Client.md]: Absolutely. I just added it.

**Comment by ShogunPanda** [comment]
@mcollina @ronag I'm not sure CI failure are related to my changes. Can you please doublecheck?

File: test/fetch/request.js
Comment (KhafraDev) [inline test/fetch/request.js:18]: This might be related to the v16.8 CI failure,
```
     # Subtest: pre aborted signal
        ok 1 - should be equal
        not ok 2 - should be equal
          ---
          found: null
          wanted: gwak
          compare: ===
          at:
            line: 285
            column: 7
            file: test/fetch/request.js
            type: Test
          stack: |
            Test.<anonymous> (test/fetch/request.js:285:7)
          source: |2
              if (hasSignalReason) {
                t.equal(req.signal.reason, 'gwak')
            ------^
              }
              t.end()
          ...
        
        1..2
        # failed 1 of 2 tests
    not ok 16 - pre aborted signal # time=14ms
```

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
This seems to break Node.js v16.8 (unfortunately we need to keep supporting this).

**Comment by ShogunPanda** [comment]
I'll double check this on Mon to fix on 16.

File: test/fetch/request.js
Comment (ShogunPanda) [reply test/fetch/request.js:18]: That made the trick, thanks!

**Comment by ShogunPanda** [comment]
The CI passes now on 16.
On 19 it passes on my machine and I cannot even find the failing test in the CI logs.

@mcollina @KhafraDev What shall I do here?

**Comment by SimenB** [comment]
Tap output isn't the easiest to skim, but you can search for `not ok`

image

**Comment by ShogunPanda** [comment]
Thanks @SimenB!
I knew and I already did it: it only returns the failing test file but apparently all tests passes inside it.

Am I wrong?

**Review by KhafraDev — APPROVED** [review]
that test is flaky

**Comment by ShogunPanda** [comment]
Ok, so I'll leave it to you to eventually re-run the CI and/or merge this if we're ready. Thanks!

**Comment by mcollina** [comment]
Can you rebase on top of main?

**Comment by mcollina** [comment]
Actually no need

**Pull request merged**
